### PR TITLE
fix(single-spa): fix for the ui not loading in Safari

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Fix a regex that was causing the UI to not display in Safari.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Run the app with `yarn start` so that you are viewing the site with single-spa.
- Open the ui in Safari, it should load.
- Double check that it still works in Firefox/Chrome.

## Fixes

Fixes: #3045.

### CODE REVIEW

**Comment by huwshimi** [comment]
@hatched maybe would you mind having a look at this as you've got Safari?

File: ui/src/single-spa-entry.js [hunk 1]
```diff
@@ -27,10 +27,18 @@ const reactLifecycles = singleSpaReact({
 
 export const { bootstrap } = reactLifecycles;
 export const mount = (props) => {
-  // Get the full path, querystring and hash. The regex gets the first
-  // forward slash that is not a double forward slash and everything after it.
-  const matches = window.location.href.match(/(?<!\/)\/(?!\/).+/);
-  // The app should never reach this entrypoint witout the basename and react
+  // Get the full path, querystring and hash (location.pathname can't be used as
+  // it does not contain the querystring and hash).
+  const matches = window.location.href
```
Comment (hatched) [inline ui/src/single-spa-entry.js:32]: You could make this a bit easier to understand for those that don't know regex by using a string replace to remove the protocol and hostname. 

```js
const loc = window.location;
const matches = loc.href.replace(`${loc.protocol}//${loc.host}`, '');
```
or you can reconstruct it
```js
const loc = window.location;
const matches = loc.pathname + loc.search + loc.hash;
```

**Review by hatched — APPROVED** [review]
Implementers choice :)

**Comment by hatched** [comment]
QA OK in FF and Safari.

File: ui/src/single-spa-entry.js (hunk 1 above)
Comment (huwshimi) [reply ui/src/single-spa-entry.js]: I've done this and it's a lot cleaner, would you mind doing another review/QA?

**Review by hatched — APPROVED** [review]

**Comment by hatched** [comment]
Looks good, thanks for the updates!

**Pull request merged**